src/Rules/Data.hs

```diff
@@ -29,6 +29,13 @@ buildPackageData context@Context {..} = do
         -- GhcCabal may run the configure script, so we depend on it.
         whenM (doesFileExist $ configure <.> "ac") $ need [configure]
 
+        -- pkgs not requiring inplaceLibCopyTargets to build
+        ghcCabalDeps <- pkgDependencies ghcCabal
+        deriveConstantsDeps <- pkgDependencies deriveConstants
+        let pkgs = ghcCabal : deriveConstants : (ghcCabalDeps ++ deriveConstantsDeps)
+        when (not $ package `elem` pkgs) $ do
```

Comment:
You could save a `not` by ``unless (package `elem` pkgs) $ need inplaceLibCopyTargets``.